### PR TITLE
add recipe for r-getip, using cran_r_skeleton_helper

### PR DESCRIPTION
**Pull request by brendanf**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by brendanf** [comment]
Split out from #24498

File: recipes/r-getip/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+source:
+  url:
+    - {{ cran_mirror }}/src/contrib/getip_{{ version }}.tar.gz
+    - {{ cran_mirror }}/src/contrib/Archive/getip/getip_{{ version }}.tar.gz
+  sha256: 0ec1677309dc3f2d861d61e7b3de0e74d1fc94aefd03bc5a6c27059e30f3cc24
+
+build:
+  merge_build_host: True  # [win]
+  number: 0
```
Comment (mfansler) [inline recipes/r-getip/meta.yaml:17]: Fix DSO issue (following [proposed fix](https://github.com/conda/conda-build/pull/4786)):

```suggestion
  missing_dso_whitelist:
    - '*/R.dll'           # [win]
  number: 0
```

**Review by mfansler — CHANGES_REQUESTED** [review]

**Comment by mfansler** [comment]
> I would expect that `R.dll` should be packaged by `r-base`, but it somehow isn't being found in this check (although the actual `R CMD install` seems to have worked fine). I can't see that `getip` is doing anything unconventional with library paths. Anyone have any advice?

The linking analysis doesn't check the R-specific library path, so we have to whitelist it. Similarly, can come up for `libRblas.dll` and `libRlapack.dll`.

**Comment by brendanf** [comment]
I notice that the linter now complains about `conda-forge/r` in the maintainer list.  Is that something that needs to be addressed?

**Comment by brendanf** [comment]
@conda-forge/r Should be ready to go.  Thanks @mfansler for the fix.

**Comment by mfansler** [comment]
> I notice that the linter now complains about `conda-forge/r` in the maintainer list. Is that something that needs to be addressed?

No, [that's a bug](https://github.com/conda-forge/conda-smithy/issues/1817).

**Review by mfansler — APPROVED** [review]

**Comment by mfansler** [comment]
Thanks for submitting this! ❤️

**Pull request merged**